src/Raven.Server/Documents/Handlers/Processors/Subscriptions/SubscriptionsHandlerProcessorForCreateSubscription.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Handlers.Processors.Subscriptions;
+
+internal class SubscriptionsHandlerProcessorForCreateSubscription : AbstractSubscriptionsHandlerProcessorForCreateSubscription<SubscriptionsHandler,
+    DocumentsOperationContext>
+{
+    public SubscriptionsHandlerProcessorForCreateSubscription([NotNull] SubscriptionsHandler requestHandler) : base(requestHandler)
+    {
+    }
+    protected override Task<ChangeVectorValidationResult> TryValidateChangeVectorAsync(SubscriptionsHandler requestHandler,
+        TransactionOperationContext context, SubscriptionCreationOptions options, SubscriptionConnection.ParsedSubscription sub)
+    {
+        return TryValidateChangeVector(requestHandler, context, options, sub);
+    }
+
+    protected override Task<(long Index, long SubscriptionId)> CreateSubscriptionAsync(SubscriptionsHandler requestHandler, string databaseName, long? id, bool? disabled, SubscriptionCreationOptions options,
+        ChangeVectorValidationResult changeVectorValidationResult)
+    {
+        return CreateSubscription(requestHandler, databaseName, id, disabled, options, changeVectorValidationResult);
+    }
+
+    protected override ValueTask GetResponsibleNodesAndWaitForExecutionAsync(SubscriptionsHandler requestHandler, string name, long index)
+    {
+        return GetResponsibleNodesAndWaitForExecution(requestHandler, name, index);
+    }
+
+    protected override void ThrowOnRevisionsIfNeeded(bool subRevisions)
+    {
+        // no-op
+    }
+
+    internal static Task<ChangeVectorValidationResult> TryValidateChangeVector(SubscriptionsHandler requestHandler, TransactionOperationContext context, SubscriptionCreationOptions options, SubscriptionConnection.ParsedSubscription sub)
```

Comment:
TryValidateChangeVectorAsync?

Why is this method async anyway?

Reply:
done